crates/sncast/src/starknet_commands/declare.rs

```diff
@@ -33,41 +32,32 @@ pub struct Declare {
     pub nonce: Option<FieldElement>,
 }
 
+pub struct BuildConfig {
+    pub scarb_toml_path: Option<Utf8PathBuf>,
+    pub json: bool,
```

Comment:
Why `json` argument was added here?

Reply:
We inherit stdio here (which have been suppressed before), so we need to set the output format to json on Scarb as well.

Reply:
My question is, why it is now configurable?

Reply:
Always has been - it's a cli argument https://github.com/foundry-rs/starknet-foundry/blob/03bbebdc6abd70db2a0ecb1acbd9b00ab7aded4a/crates/sncast/src/main.rs#L63-L65

It just has never been passed to Scarb, as Scarb output has been suppressed anyway.